file/writable_file_writer.cc

```diff
@@ -580,9 +588,12 @@ IOStatus WritableFileWriter::WriteBufferedWithChecksum(const char* data,
   if (rate_limiter_ != nullptr) {
     while (data_size > 0) {
       size_t tmp_size;
-      tmp_size = rate_limiter_->RequestToken(
-          data_size, buf_.Alignment(), writable_file_->GetIOPriority(), stats_,
-          RateLimiter::OpType::kWrite);
+      Env::IOPriority rate_limiter_priority_used =
+          WritableFileWriter::DecideRateLimiterPriority(
```

Comment:
Note: 
There might be an optimization opportunity to move `Env::IOPriority rate_limiter_priority_used =
WritableFileWriter::DecideRateLimiterPriority` outside `while (data_size > 0)`  in `WriteBuffered/DirectWithChecksum`. But it's also prone to a hypothetical error where the the result of `writable_file_->GetIOPriority()` changes between the passes of loop (seems rarely but we never know for the future). 

Considering the `WriteBuffered/DirectWithChecksum()`  (i.e, assert(perform_data_verification_ && buffered_data_with_checksum_);) are only for Manifest and WAL now based on the original [PR](https://github.com/facebook/rocksdb/pull/8412) and code inspection AND none of them are affected by this code change since they are current not rate-limited, I decided not to pursue this optimization in this PR. Instead, I will benchmark it in my next PR of rate-limiting WAL (https://github.com/facebook/rocksdb/pull/9607) to see if there is any serious perf regression of my current impl first.